### DISCUSSION TITLE
Problem with json request

### DISCUSSION THREAD

**Discussion by Kolhun** [curl/curl#11719] [Q&A]
Hello everyone, I'm trying to make a json request to my server (screen one) and gives an error that usually gives out if the request is made incorrectly (screen 3). At the same time, if I make a request from the browser plugin, everything works fine (screen 2). The TOKEN is disguised as "??????" for security purposes. json.dump().c_str() returns a string. What could be the problem?

![image](https://github.com/curl/curl/assets/36815965/b0a1deb6-055f-478e-942e-5dfed8f90718)
![image](https://github.com/curl/curl/assets/36815965/e78ba7c4-d180-4022-b5af-7a0a37c1d726)
![image](https://github.com/curl/curl/assets/36815965/3d6e30b0-9d22-4587-9072-9f51b059c6eb)

**Comment by emanuele6** [comment]
Wouldn't `headers.dump().c_ptr()` return a string like `"{\"Content-Type\":\"application/json\",\"TOKEN\":\"?????\"}"`? Appending that to a `curl_slist` and then using that list as headers does not make much sense.

You should append to the string list two elements formatted as `name: value` (like a HTTP header), not a single string that is a JSON object.

```c++
curl_slist *tmp;
curl_slist *headers = curl_slist_append(NULL, "Content-Type: application/json");
// if (!headers) handle_error();
tmp = curl_slist_append(headers, "TOKEN: ?????");
// if (!tmp) handle_error();
headers = tmp;

curl_easy_setopt(curl, CURL_HTTPHEADER, headers);

// perform

curl_slist_free_all(headers);
```

**Reply by Kolhun** [reply]
I tried to do this, now the token sees, but json does not digest something
![image](https://github.com/curl/curl/assets/36815965/b151a09a-317b-49f1-98f1-96980458838e)
![image](https://github.com/curl/curl/assets/36815965/6bd20e32-9198-4b21-8e52-f20e5651cf6f)

**Reply by Kolhun** [reply]
![image](https://github.com/curl/curl/assets/36815965/641e368d-2e8b-4388-b4fd-26504350b0a2)

**Comment by Kolhun** [comment]
I have solved the issue
